pkg/common-controller/snapshot_controller_base.go

```diff
@@ -215,9 +215,16 @@ func (ctrl *csiSnapshotCommonController) snapshotWorker() {
 			// The volume snapshot still exists in informer cache, the event must have
 			// been add/update/sync
 			newSnapshot, err := ctrl.checkAndUpdateSnapshotClass(snapshot)
-			if err == nil {
+			if err == nil || (newSnapshot.ObjectMeta.DeletionTimestamp != nil && errors.IsNotFound(err)) {
+				// If the VolumeSnapshotClass is not found, we still need to process an update
+				// so that syncSnapshot can delete the snapshot, should it still exist in the
+				// cluster after it's been removed from the informer cache
 				klog.V(5).Infof("passed checkAndUpdateSnapshotClass for snapshot %q", key)
 				ctrl.updateSnapshot(newSnapshot)
+			} else if newSnapshot.ObjectMeta.DeletionTimestamp != nil && errors.IsNotFound(err) {
+
+				klog.V(5).Infof("snapshotClass not found and snapshot %q is scheduled for deletion", key)
+				ctrl.updateSnapshot(newSnapshot)
```

Comment:
This is already handled by the code block at 218-223.

Reply:
You're correct. I accidentally left this in; it should now be removed.